### ISSUE TITLE
The loss of training fine-tune dinov3 is nan

### ISSUE THREAD

**Issue by mcmingchang** [tue-mps/eomt#47]
The loss of training fine-tune dinov3 is nan.
python main.py fit -c configs/dinov3/coco/instance/eomt_large_640.yaml --trainer.devices 4 --data.batch_size 4 --model.ckpt_path model_zoo/dinov3.bin --model.load_ckpt_class_head False --model.delta_weights False --data.path train_data/snack_seg

But training from scratch according to the following instructions is not a problem.
python main.py fit -c configs/dinov3/coco/instance/eomt_large_640.yaml --trainer.devices 4 --data.batch_size 4  --data.path train_data/snack_seg

Dinov2 training is also fine.
python main.py fit -c configs/dinov2/coco/instance/eomt_large_640.yaml --trainer.devices 4 --data.batch_size 4 --model.ckpt_path model_zoo/dinov2.bin --model.load_ckpt_class_head False --data.path train_data/snack_seg

**Comment by NiccoloCavagnero** [comment]
Indeed `_zero_init_outside_encoder` should only be called when `delta_weights=True` and `load_ckpt_class_head=False`. 

With respect to a lower initial performance with respect to DINOv2, we experienced a similar thing when training the DINOv3 models, the last layers of the DINOv3 are delicate.

You can try to adapt the schedule more by increasing warmup steps and adopting a better learning rate.

Best,
Nick

**Comment by NiccoloCavagnero** [comment]
Hi @mcmingchang,

When commenting that out you are not properly loading the EoMT checkpoint, in particular queries and heads are not correctly initialized.

Could you try using bf16 in place of fp16?

Best,
Nick
